main.cpp

```diff
@@ -28,7 +28,7 @@ EventFlags threadFlag;
 
 void print_socket_stats()
 {
-    mbed_stats_socket_t stats[MBED_CONF_NSAPI_SOCKET_STATS_MAX_COUNT] = {0};
+    static mbed_stats_socket_t stats[MBED_CONF_NSAPI_SOCKET_STATS_MAX_COUNT];
```

Comment:
I believe this is arguably better anyway - should probably avoid putting potentially-large objects on the stack in case of overflow.

Although on the other hand this *is* just a temporary for the print, so if there was no overflow danger, automatic is good.

If wanting to keep automatic, a better (C++11 or later) syntax to zero-init than `= { 0 }` is just `{}` (no `=`). That is probably simpler for the compiler, and could also avoid the bug.

Reply:
What was the assertion failure?

Reply:
Cryptic error:

```
[DEBUG] Return: 3
[DEBUG] Output: Internal error: [Front end]: assertion failed at: "/home/jenkins_ci01/slave_top/workspace/ArmReleaseBuild/core/translator/compiler_core/src/parser/edg/il.c", line 23626
```

Reply:
I also tested `mbed_stats_socket_t stats[MBED_CONF_NSAPI_SOCKET_STATS_MAX_COUNT] {};` , errors as well